tests/providers/google/cloud/utils/gcp_authenticator.py

```diff
@@ -107,8 +107,8 @@ def set_key_path_in_airflow_connection(self):
             extras[PROJECT_EXTRA] = self.project_extra if self.project_extra else self.project_id
             conn.extra = json.dumps(extras)
             session.commit()
-        except BaseException as ex:
-            self.log.error('Airflow DB Session error: %s', str(ex))
+        except BaseException:
+            self.log.error('Airflow DB Session error.')
```

Comment:
I thought that the `raise` was sufficient here for the additional traceback information which includes the exception message. Generally I was thinking to keep `logging.error()` when there was an accompanying `raise`, but use `logging.exception()` when there isn't. Happy to hear other thoughts of course!

Reply:
But perhaps going this route for core and provider modules, we get into a situation where log handlers don't output the same (possibly useful?) information. I'd love feedback here.

Reply:
Same here. we do not need to catch BaseException here, we just need a fixture that calls rollback() (it is safe to call  rollback even after commit) . then we do not heed to have try/except at all.

Reply:
This log really adds no value here.

After looking at at that - It should be REALLY ok to just remove the except clause altogether. It's not needed here at all. 


The session.close() method in finally will automatically rollback the transaction, there is no need to do it manually in "except" clausehttps://docs.sqlalchemy.org/en/14/orm/session_basics.html#closing:

> The close() method issues a expunge_all(), and [releases](http://docs.sqlalchemy.org/en/latest/glossary.html#term-releases) any transactional/connection resources. When connections are returned to the connection pool, transactional state is rolled back as well.

Reply:
Being a SQLAlchemy novice here, but it looks like using a context manager here for the session object should suffice? And the `session.commit()` doesn't seem necessary since there is only read activity with the session. Unless I'm missing something looking at the docs.

Reply:
Yep. That should do :)